packages/devtools_app/lib/src/charts/flutter/flame_chart.dart

```diff
@@ -71,21 +252,52 @@ class FlameChartNode<T> extends StatelessWidget {
         preferBelow: false,
         child: InkWell(
           onTap: () => onSelected(data),
-          child: Container(
-            padding: const EdgeInsets.only(left: 6.0),
-            alignment: Alignment.centerLeft,
-            color: selected ? _selectedNodeColor : backgroundColor,
-            child: Text(
-              text,
-              textAlign: TextAlign.left,
-              overflow: TextOverflow.ellipsis,
-              style: TextStyle(
-                color: selected ? Colors.black : textColor,
-              ),
-            ),
+          child: ValueListenableBuilder(
+            valueListenable: selectionListenable,
```

Comment:
My understanding of this is that all widgets in the chart will listen to the listenable.  

This will rebuild every flame chart node when the listenable notifies, which is going to be the same as just rebuilding the chart.

If the listenable implementation is a ChangeNotifier or ValueNotifier, then notifications will even propagate in O(N²) time instead of O(N) time.